src/main/java/com/company/array/DoesSubArrayHavingSumEqualToK.java

```diff
@@ -0,0 +1,20 @@
+package com.company.array;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DoesSubArrayHavingSumEqualToK {
+    public boolean doesSubArraysHavingSumEqualToK(int[] nums, int k) {
+        List<Integer> list = new ArrayList<>();
+        int sum = 0;
+        list.add(0);
+        for (int num : nums) {
+            sum += num;
+            if (list.contains(sum - k)) {
```

Comment:
Can we use the HashSet here so that this code will be optimised to O(N)? Currently your code has O(N^2) time complexity because of using list.contains method inside for loop.

Reply:
Yeah sure.. changed to HashSet